### ISSUE TITLE
Rewrite channel/frequency conversions and handling

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1839]
A few things to keep in mind:
- Some wireless cards can go further than the normal channel list
- libnl (only available on Linux) deals with frequency mostly
- Conversion is also needed to display stuff to the user, so we'll need tables for different frequency bands (900MHz, 3.6GHz, 2.4GHz, 5GHz and 60GHz) and [Wikipedia](https://en.wikipedia.org/wiki/List_of_WLAN_channels#5_GHz_(802.11a/h/j/n/ac/ax)) has a list.
- In case there isn't a mapping, then just display frequency.
- Handle the case of channel width: 5/10/20/40/80/160

Open to discussion.

**Comment by psyborg55** [comment]
include 2.2-2.7GHz in 2.4 band, 4.9-6.1GHz in 5 band
reduce scan time in airodump

**Comment by psyborg55** [comment]
https://github.com/aircrack-ng/aircrack-ng/issues/1928

**Comment by aircrack-ng** [comment]
Assuming scan time is channel hopping time, this can already be changed with -f

**Comment by psyborg55** [comment]
no, scan time when launching airodump that checks available freq, it's output is:
Checking available frequencies, this could take few seconds.

on a system with 2312-2484 enabled it will disconnect active connection while checking available freqs

**Comment by aircrack-ng** [comment]
All the stuff you mention is already in the list above.

Regarding disconnecting the active connection, that is your role, using airmon-ng

**Comment by ZeroChaos-** [comment]
2.192GHz - 2.732GH and 4.8GHz - 6.1 GHz is my personal best, which I think is already reflected by "airodump-ng -C0"

**Comment by ChillVibesMushroom** [comment]
-C parameter when used we would find the frequency using Wireshark right then we check headers to find the right frequency?
